app/src/main/java/io/appium/uiautomator2/model/UiObject2Element.java

```diff
@@ -98,77 +100,107 @@ public String getText() {
         return element.getText() != null ? element.getText() : "";
     }
 
+    @Override
     public String getName() {
         return element.getContentDescription();
     }
 
-    public String getStringAttribute(final String attr) throws NoAttributeFoundException {
-        if ("name".equalsIgnoreCase(attr) || "text".equalsIgnoreCase(attr)) {
-            return getText();
-        }
-        if ("contentDescription".equalsIgnoreCase(attr)) {
-            return element.getContentDescription();
-        }
-        if ("className".equalsIgnoreCase(attr)) {
-            return element.getClassName();
-        }
-        if ("resourceId".equalsIgnoreCase(attr) || "resource-id".equalsIgnoreCase(attr)) {
-            return element.getResourceName();
+    @Nullable
+    @Override
+    public String getAttribute(String attr) throws NoAttributeFoundException, UiObjectNotFoundException {
+        if (attr == null) {
+            throw generateNoAttributeException("null");
         }
-        throw generateNoAttributeException(attr);
-    }
 
-    public boolean getBoolAttribute(final String attr) throws UiAutomator2Exception {
-        switch (attr) {
+        final Object result;
+        switch (attr.toLowerCase()) {
```

Comment:
Other than the `"displayed"` attribute, this looks the same as in `UiObjectElement.java`. Is there no way to make this shared code? In fact much of this class seems the same as that one.

Reply:
yep I was also thinking about that. Unfortunately UiObject and UiObject2 don't share any common interfaces. The only way here is to use reflection, which I'm not very happy with.

Reply:
But both the classes implement `AndroidElement`. Is there no way to have an abstract class that they both extend, which implements the common stuff?

Reply:
this will require to create a separate getter for each attribute in the main abstract class and override it in both inherited classes, which is even more code

Reply:
Ugh. Ok.